core/src/header.rs

```diff
@@ -244,3 +244,26 @@ impl HeaderBuilder {
         self.inner
     }
 }
+
+#[derive(Clone, Serialize, Deserialize, PartialEq, Eq, Debug, Default)]
+pub struct RichHeader {
```

Comment:
Weird name... need documentation or better naming.

Reply:
I still don't understand why to define a new struct in core mod with a weird name, is it only used in sync mod?  can we move it to sync mod and just use a type alias?

Reply:
@zhangsoledad is working on this.